### PR TITLE
Higadrite Metabolization Fix

### LINKED ISSUE #49512: Microdosages of certain chems cause effects to never wear off
## Round ID:
N/A (this was on local)

## Reproduction:
Apply a dosage of higadrite to yourself below the metabolism rate. Enjoy the infinite sturdy liver.
Best guess is that it metabolises without proccing the "on end metabolise" block, so the trait never gets cleared.

### PR DESCRIPTION
**Pull request by RiskySikh**
## About The Pull Request
Fixes #49512 

There has been issues regarding the chemical higadrite, since the chemical doesn't disappear, and keeps on metabolizing. The issue was caused by the on_mob_add that keeps on adding the chemical, without it stopping. Which is why it is replaced with on_mob_metabolize.

## Why It's Good For The Game

Well, it fixes the exploit which is that it keeps giving the liver infinite protection from toxin damage.

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -1060,7 +1060,7 @@
 	color = "#FF3542"
 	self_consuming = TRUE
 
-/datum/reagent/medicine/higadrite/on_mob_add(mob/living/M)
+/datum/reagent/medicine/higadrite/on_mob_metabolize(mob/living/M)
 	..()
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]:
```suggestion
	. = ..()
```

**Comment by ATH1909** [comment]
Will on_mob_metabolize() still trigger if you don't have a liver for a self_consuming = TRUE chem?

**Comment by RiskySikh** [comment]
> 
> 
> Will on_mob_metabolize() still trigger if you don't have a liver for a self_consuming = TRUE chem?

nope

**Comment by ExcessiveUseOfCobblestone** [comment]
that's the whole point of the var lol are you sure?

**Comment by RiskySikh** [comment]
> 
> 
> that's the whole point of the var lol are you sure?
nevermind

**Pull request merged**